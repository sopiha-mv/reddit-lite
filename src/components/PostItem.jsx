import React from 'react';
import PropTypes from 'prop-types';

const PostItem = props => (
	<div className={'media mb-3 border align-items-stretch'}>
		{
			props.thumbnail !== 'self' && 
				<img className={'align-self-start'} src={props.thumbnail} />
		}
		<div className={'media-body pt-3 pr-3 pl-3'}>
			<div className={'d-flex flex-column h-100'}>
				<h5 className={'mt-0'}>{props.title}</h5>
				<div className={'mt-auto d-flex justify-content-between'}>
					<p>{props.author}, {props.time}</p>
					<a href={props.postLink}>read more</a>
				</div>
			</div>
		</div>
	</div>
);

PostItem.propTypes = {
	title: PropTypes.string,
	postLink: PropTypes.string,
	thumbnail: PropTypes.string,
	author: PropTypes.string,
	time: PropTypes.string,
};

export default PostItem;