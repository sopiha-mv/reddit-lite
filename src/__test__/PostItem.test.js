import React from 'react';
import PostItem from '../components/PostItem.jsx';
import renderer from 'react-test-renderer';

it("render correctly post component", () => {
	const component = renderer.create(
		<PostItem
			title={'mockTitle'}
			postLink={'mockPostLink'}
			thumbnail={'mockThumbnail'}
			author={'mockAuthor'}
			time={'mockTime'}
		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});