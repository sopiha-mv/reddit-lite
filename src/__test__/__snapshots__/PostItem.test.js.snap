// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly post component 1`] = `
<div
  className="media mb-3 border align-items-stretch"
>
  <img
    className="align-self-start"
    src="mockThumbnail"
  />
  <div
    className="media-body pt-3 pr-3 pl-3"
  >
    <div
      className="d-flex flex-column h-100"
    >
      <h5
        className="mt-0"
      >
        mockTitle
      </h5>
      <div
        className="mt-auto d-flex justify-content-between"
      >
        <p>
          mockAuthor
          , 
          mockTime
        </p>
        <a
          href="mockPostLink"
        >
          read more
        </a>
      </div>
    </div>
  </div>
</div>
`;
